### ISSUE TITLE
HTTP Tool Header Auth Not Working As Expected

### ISSUE THREAD

**Issue by jimleuk** [n8n-io/n8n#9971]
### Bug Description

Attempting to use Header Auth credentials for the AI Agent's HTTP tool fails with the following error:
```
{
  "response": "There was an error: \"Cannot set properties of undefined (setting 'Authorization')\""
}
```
When I define the header auth manually using the "Send Headers" options, the HTTP tool works as expected.

The service I was using is the Firecrawl.dev API. My intention is to use the HTTP tool scrape webpages.

<details><summary>Workflow.json Attached Here</summary>
<p>
{
  "meta": {
    "instanceId": "26ba763460b97c249b82942b23b6384876dfeb9327513332e743c5f6219c2b8e"
  },
  "nodes": [
    {
      "parameters": {},
      "id": "c0a6afed-a527-45cf-8bff-bcb51628c6d4",
      "name": "When chat message received",
      "type": "@n8n/n8n-nodes-langchain.chatTrigger",
      "typeVersion": 1,
      "position": [
        1040,
        360
      ],
      "webhookId": "e11e6a70-32c3-434d-9f40-0154a61b918d"
    },
    {
      "parameters": {
        "toolDescription": "Call this tool to fetch a webpage content.",
        "method": "POST",
        "url": "https://api.firecrawl.dev/v0/scrape",
        "authentication": "genericCredentialType",
        "genericAuthType": "httpHeaderAuth",
        "sendBody": true,
        "parametersBody": {
          "values": [
            {
              "name": "url"
            },
            {
              "name": "pageOptions",
              "valueProvider": "fieldValue",
              "value": "={{ {\n  onlyMainContent: true,\n  replaceAllPathsWithAbsolutePaths: true,\n  removeTags: 'img,svg,video,audio'\n} }}"
            }
          ]
        },
        "optimizeResponse": true,
        "dataField": "data",
        "fieldsToInclude": "selected",
        "fields": "markdown"
      },
      "id": "fb4abae8-7e38-47b7-9595-403e523f7125",
      "name": "HTTP Webscraper Tool",
      "type": "@n8n/n8n-nodes-langchain.toolHttpRequest",
      "typeVersion": 1,
      "position": [
        1460,
        540
      ],
      "credentials": {
        "httpHeaderAuth": {
          "id": "OUOnyTkL9vHZNorB",
          "name": "Firecrawl API"
        }
      }
    },
    {
      "parameters": {
        "options": {}
      },
      "id": "9fc05c79-5a2d-4ac4-a4f5-32b9c1b385e1",
      "name": "OpenAI Chat Model",
      "type": "@n8n/n8n-nodes-langchain.lmChatOpenAi",
      "typeVersion": 1,
      "position": [
        1240,
        540
      ],
      "credentials": {
        "openAiApi": {
          "id": "8gccIjcuf3gvaoEr",
          "name": "OpenAi account"
        }
      }
    },
    {
      "parameters": {
        "options": {}
      },
      "id": "066f9cdd-4bd3-48a1-bf9b-32eda3e28945",
      "name": "AI Agent",
      "type": "@n8n/n8n-nodes-langchain.agent",
      "typeVersion": 1.6,
      "position": [
        1260,
        360
      ]
    }
  ],
  "connections": {
    "When chat message received": {
      "main": [
        [
          {
            "node": "AI Agent",
            "type": "main",
            "index": 0
          }
        ]
      ]
    },
    "HTTP Webscraper Tool": {
      "ai_tool": [
        [
          {
            "node": "AI Agent",
            "type": "ai_tool",
            "index": 0
          }
        ]
      ]
    },
    "OpenAI Chat Model": {
      "ai_languageModel": [
        [
          {
            "node": "AI Agent",
            "type": "ai_languageModel",
            "index": 0
          }
        ]
      ]
    }
  },
  "pinData": {}
}
</p>
</details> 

### With Generic Credential Header Auth

![Screenshot 2024-07-08 150944](https://github.com/n8n-io/n8n/assets/373596/74d64961-3817-4e74-a067-5739f64e32cf)
![Screenshot 2024-07-08 150839](https://github.com/n8n-io/n8n/assets/373596/0ac550f5-8a21-456f-9584-a40af5cae88f)

### Without Generic Credential Header Auth

![Screenshot 2024-07-08 151102](https://github.com/n8n-io/n8n/assets/373596/a50276f6-d5cd-4b49-900d-ae3d2a5eb770)
![Screenshot 2024-07-08 151120](https://github.com/n8n-io/n8n/assets/373596/8d5d40da-defb-4c0d-939c-f0d8a250dc5e)

### To Reproduce

1. Create an AI Tools Agent
2. Attach HTTP tool subnode
3. Configure HTTP tool subnode to make a request to service requiring "Header Auth"
4. Set Credentials as Generic Credential Type and Header Auth
5. Trigger HTTP tool to be called via AI Tools Agent.

### Expected behavior

HTTP tool is able to authorise with Header Auth credential and not error.

### Operating System

Windows 11 Home x WSL Ubuntu 

### n8n Version

1.48.0

### Node.js Version

v20.14.0

### Database

SQLite (default)

### Execution mode

main (default)

**Comment by Joffcom** [comment]
Hey @jimleuk,

Thanks for the report, I have created `AI-210` as the dev ticket for this.

**Comment by jimleuk** [comment]
Hey team, just wanted to add another test case I caught yesterday
* This time using Predefined Credential Type for HTTP tool using SERPAPI for searching flights
* SERP API uses query auth which is `?api_key=xxxxxxx`
* I get a 400 error when calling the tool and in SERPAPI search history page, I can see the search was not registered which makes me believe that this is an auth error.
* I add the query auth manually and calling the tool works as expected.

Thanks!

![Screenshot 2024-07-09 102522](https://github.com/n8n-io/n8n/assets/373596/e52f9b9d-a64a-4665-b15d-19ded45f6e43)
![Screenshot 2024-07-09 102558](https://github.com/n8n-io/n8n/assets/373596/2a51ad38-edf9-4493-a295-6aaa3659d1cb)

**Comment by janober** [comment]
Fix got released with [`n8n@1.51.0`](https://github.com/n8n-io/n8n/releases/tag/n8n%401.51.0)
